### PR TITLE
need to have some mechanism to prevent from memory corruption

### PR DESCRIPTION
**Pull request by studentBC**

### CODE REVIEW

File: StitchPanorama.py [hunk 1]
```diff
@@ -1,77 +1,118 @@
...
+            return panorama
 
-        good = []
-        for m,n in matches:
-            if m.distance < 0.03*n.distance:
-                good.append(m)
+    def prepare_lists(self):
+        self.centerIdx = self.count/2
+        self.center_im = self.images[int(self.centerIdx)]
+        for i in range(self.count):
```
Comment (chiahaol) [inline StitchPanorama.py:46]: I think there should be a way to avoid using for loop, like self.images[: self.centerIndex]

Comment (studentBC) [reply StitchPanorama.py:46]: but every frame is uniq one so you still need a for loop within it like self.images[frames[i], for i in range(self.centerIndex)]

File: StitchPanorama.py [hunk 2]
```diff
@@ -1,77 +1,123 @@
...
-            pts = np.float32([ [0,0],[0,h-1],[w-1,h-1],[w-1,0] ]).reshape(-1,1,2)
-            dst = cv2.perspectiveTransform(pts, M)
-            img2 = cv2.polylines(img2,[np.int32(dst)],True,255,3, cv2.LINE_AA)
-            #cv2.imshow("original_image_overlapping.jpg", img2)
-        else:
-            print("Not enought matches are found - %d/%d", (len(good)/MIN_MATCH_COUNT))
-            return 0, img 
+    def leftshift(self):
+        a = self.left_list[0]
+        for b in self.left_list[1:]:
```
Comment (chiahaol) [inline StitchPanorama.py:54]: I think this line will create a copy, it's better to use index to iterate

**Review by chiahaol — APPROVED** [review]

File: StitchPanorama.py (hunk 1 above)
Comment (chiahaol) [reply StitchPanorama.py:46]: self.left_list = self.images[:self.centerIndex + 1]

**Pull request merged**